docs/source/etl/loaders.rst

```diff
@@ -0,0 +1,299 @@
+
+This versatile `Loader` empowers users to create a diverse array of plots, including various forms of bar charts.
+By configuring parameters within the ETL step, users can customize the ``ColumnCrossPlotLoader`` to suit their specific needs.
+
+See `demo_project/doe-suite-config/super_etl/demo02-colcross.yml` for an example of how to configure the ``ColumnCrossPlotLoader``.
+
+Moreover, the ``ColumnCrossPlotLoader`` offers multiple hooking points, enabling further customization through the integration of custom code.
+This ensures that users have the flexibility to leverage the basic features even if they have a highly specialized requirement.
+See `demo_project/doe-suite-config/does_etl_custom/etl/plots/colcross.py` for an example of how to extend the loader with custom components.
+
+
+.. autopydantic_model:: doespy.etl.steps.colcross.colcross.ColumnCrossPlotLoader
+    :inherited-members: BaseModel
+
+
+Configurations (Cumulative)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The `ColumnCrossPlotLoader` facilitates the creation of multiple figures, potentially containing grids of subplots, each presenting a chart composed of multiple artists.
+
+To ensure maximum customization flexibility, the loader allows to configure every aspect separately.
+This is achieved through cumulative lists, that are merged to establish configurations for figure-level attributes, subplot-level attributes, and individual artists.
+
+The configurations are merged cumulatively, giving priority to those listed earlier in case of conflicts.
+Leveraging the ``jp_query`` within list entries enables the creation of specific configurations tailored to the data used for their generation.
+
+Each figure, subplot, and artist (e.g., a bar segment in a stacked bar chart) is assigned a unique identifier based on the underlying data.
+These identifiers can be used to address particular configurations to subsets of plots, subplots, and artists.
+
+For instance, subplots may have different chart types, or specific bars may require distinct colors.
+Hence, by formulating a ``jp_query`` that matches on the identifer of a specific bar, we can apply a unique color to that bar.
+
+Conceptually, the generation process is a hierarchical series of groupby operations.
+At each level, an ID is assigned -- a flat JSON/dictionary object comprising key-value pairs derived from the grouping columns of the current level and all preceding levels.
+For instance, if figures are created for each unique value in the 'year' column, resulting identifiers follow the format ``plot_id = {'year': 2000}`` or ``plot_id = {'year': 2001}``.
+
+In addition to the grouping columns, IDs include supplementary information, such as the row and column indices of a subplot within a grid.
+
+Below we show the hierarchical structure of the IDs, starting from the top-level figure and descending to the lowest-level artist:
```

Comment:
I did not fully understand this example

Reply:
Can you explain what you understand and what is unclear?

The loader allows the creation of multiple figures. Namely, it will create a figure for every unique combination of values in a defined set of columns. For example, `cols: [system]` means to create a figure for every value in the `system` column. 
If you would implement this logic yourself, given a pandas data frame, you would probably do something like (in pseudocode):

```
for df1 in df.groupby(cols=[system]):
    fig = plt.figure()
    fig.plot(...df1...)
    fig.savefig(...)
```


Now because we not only create multiple figures, but also each figure can consist of a grid of subplots, and each subplot can e.g., be a GroupedBarChart, we repeat this same idea:


```py
for df_fig in df.groupby(cols=[system]): # fig_foreach
     for df_subfig in df_fig.groupby(cols=[...]): # subplot_foreach
          for df_group in df_subfig.groupby(cols=[...]): # group_foreach
               # df_group contains only the data of a group of bars min a subfigure in a figure
               for df_bar in df_group.groupby(cols=[...]) :  # bar_foreach (in the example below called element-level)
                    for df_part in df_bar.groupby(cols=[]) : # part_foreach
                        # df_part now only contains the data for a specific segment in a stacked barchart (we call this artist)
                        ax.plot(...df_part...) # just plot one segment in a grouped stacked bar chart
```